### DISCUSSION TITLE
Strange many to many behavior

### DISCUSSION THREAD

**Discussion by cah4a** [directus/directus#12254] [Archive]
I have request like `/items/users?fields=contacts.friend_user_id.name`
In my case contacts table has many rows with user_id = null.

I solve that using `?deep[contacts][_filter][friend_user_id][_neq]=null` from API point of view, but the app is still showing me every row from the contacts table.

Are there any ideas on how to force table view correctly filter empty junction rows?

**Comment by rijkvanzanten** [comment]
> Are there any ideas on how to force table view correctly filter empty junction rows?

I don't fully understand the question.. Are you asking on how to configure the filters in the app to only show items that don't have related m2m items?

**Reply by cah4a** [reply]
I'm asking on how to configure the filters in the app to only show items that HAVE related m2m items.

For the following conjunction table

|  user_id  |  phone  |  user_friend_id  |
| --- | --- | --- |
| 1  | 123123 | 4 |
| 2  | 678678 | NULL |
| 3  | 101010 | NULL |

the app shows 3 m2m items instead of 1.

**Reply by rijkvanzanten** [reply]
Ah gotcha. There's currently no way to filter based on the number of related items yet

**Reply by cah4a** [reply]
Pity.
Maybe you could suggest some workaround?

**Reply by rijkvanzanten** [reply]
Ehh I don't really know what the workaround would be. You could fetch the related collection with a filter that negates on the many to one key maybe?

**Comment by azrikahar** [comment]
This is now made possible thanks to #12488 👍 So you can do something like `?filter[count(your_related_field)][_eq]=0` for ones _without_ related items.
